frontend/dashboard/app/auth/cookie.ts

```diff
@@ -1,38 +1,40 @@
-import { NextResponse } from "next/server";
+import { NextResponse } from "next/server"
 
 export function setCookiesFromJWT(accessToken: string, refreshToken: string, response: NextResponse<any>): NextResponse<any> {
     // Decode the access token to extract the expiry
     const accessTokenPayload = JSON.parse(
         Buffer.from(accessToken.split('.')[1], 'base64').toString()
-    );
+    )
     const accessExp = new Date(accessTokenPayload.exp * 1000)
 
     // Decode the refresh token to extract the expiry
     const refreshTokenPayload = JSON.parse(
         Buffer.from(refreshToken.split('.')[1], 'base64').toString()
-    );
+    )
     const refreshExp = new Date(refreshTokenPayload.exp * 1000)
 
     // extract the domain from the NEXT_PUBLIC_SITE_URL
     // and set it as the domain for the cookies
     const domain = new URL(process.env.NEXT_PUBLIC_SITE_URL!).hostname
 
+    const isDev = process.env.NODE_ENV === 'development'
+
     response.cookies.set('access_token', accessToken, {
         path: '/',
         domain: domain,
         maxAge: Math.floor((accessExp.getTime() - Date.now()) / 1000),
         httpOnly: true,
-        secure: true,
-        sameSite: 'strict',
+        secure: !isDev,
+        sameSite: isDev ? 'lax' : 'strict',
     })
 
     response.cookies.set('refresh_token', refreshToken, {
         path: '/',
         domain: domain,
         maxAge: Math.floor((refreshExp.getTime() - Date.now()) / 1000),
         httpOnly: true,
-        secure: true,
-        sameSite: 'strict',
+        secure: !isDev,
```

Comment:
@anupcowkur All browsers, including Safari should ignore `secure: true` for local context (localhost). So, you can leave it to `true` as was before. Should work.

![image](https://github.com/user-attachments/assets/e16d9a92-3165-4896-b0f2-6e87a54f723f)

https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Set-Cookie#secure